src/zeusops_bot/cogs/zeus_upload.py

```diff
@@ -1,20 +1,53 @@
 """zeus_upload extension"""
 
+import json
+from typing import Annotated
+
 import discord
+
+# from discord.commands import option
 from discord.ext import commands
+from discord.utils import basic_autocomplete
+from pydantic import TypeAdapter, ValidationError
 
 from zeusops_bot.errors import (
     ConfigFileInvalidJson,
     ConfigFileNotFound,
     ConfigPatchingError,
 )
+from zeusops_bot.models import ModDetail
 from zeusops_bot.reforger_config_gen import ReforgerConfigGenerator
+from zeusops_bot.settings import ZeusopsBotConfig
+
+OptionalDiscordAttachment = Annotated[
+    discord.Attachment,
+    discord.Option(discord.SlashCommandOptionType.attachment, required=False),
+]
+
+modlist_typeadapter = TypeAdapter(list[ModDetail])
+
+
+async def _autocomplete_missions(ctx: discord.AutocompleteContext) -> list[str]:
+    return [
+        mission
+        for mission in ReforgerConfigGenerator.get_config().list_missions()
```

Comment:
Don't like we're not passing in the reforger config generator to get the list of missions from it, that's fucked up, the get_config() is basically a constructor.

Reply:
I couldn't figure out how to pass the config in a better way. The callback function is either passed in a decorator or a type hint, both of which are parsed early(?) before we even have initialised the config object. Would definitely prefer something that does use a pseudo-global, if we can figure out how.